postgis/postgis.sql.in

```diff
@@ -5842,7 +5842,7 @@ CREATE OR REPLACE FUNCTION ST_3DDFullyWithin(geom1 geometry, geom2 geometry,floa
 
 CREATE OR REPLACE FUNCTION _ST_3DIntersects(geom1 geometry, geom2 geometry)
```

Comment:
You're right, I thought GEOS' ST_3DIntersects did not handle TIN and PS, but since it is implemented thanks to a mindistance, it actually works with TIN and PS. So ok for ST_3DIntersects renamed to 3DSurfaceIntersects, and coming from plain LWGEOM (no SFCGAL needed here)

However for construction functions (Intersection, Difference and Union), do you agree with the proposition to split the former SFCGAL 3DIntersection into 3DSurfaceIntersection and 3DSolidIntersection ?

Reply:
I believe SurfaceIntersection and SolidIntersection are different matter. I don't believe we have a good implementation reason to split them - they can work fine with Solid flag. The intersects rename is about being mindistance internally mostly - it can't handle Solid flag properly, so we are signaling about it in the name. If we can set and unset solid from SQL, then we can live with just plain working ST_3DIntersection and not break it for everyone - there's no attempt to implement it in LWGEOM.

Reply:
In my perfect world, we can also:

 - somehow teach minsidance3d about solids and their internals, and drop SFCGAL 3DIntersects completely without unpairing them into two distinct names - we won't break API this way;

 - on non-solid intersects benchmark SFCGAL and LWGEOM versions and if LWGEOM mindist wins, short-circuit it.

(I really hope I do not sound as SFCGAL hater to you :)